scripts/basic_youtube_upload.py

```diff
@@ -0,0 +1,149 @@
+# For more information about the client_secrets.json file format, see:
+#   https://developers.google.com/api-client-library/python/guide/aaa_client_secrets
+CLIENT_SECRETS_FILE = 'client_secret.json'
+YOUTUBE_CREDENTIALS_FILE = '.youtube-upload-credentials.json'
+# This OAuth 2.0 access scope allows an application to upload files to the
+# authenticated user's YouTube channel, but doesn't allow other types of access.
+SCOPES = ['https://www.googleapis.com/auth/youtube.upload', 'https://www.googleapis.com/auth/youtube.force-ssl']
+API_SERVICE_NAME = 'youtube'
+API_VERSION = 'v3'
+
+VALID_PRIVACY_STATUSES = ('public', 'private', 'unlisted')
+
+
+# Create client from stored authorization credentials.
+def get_youtube_client():
+    credentials = google.oauth2.credentials.Credentials.from_authorized_user_file(YOUTUBE_CREDENTIALS_FILE)
+    return build(API_SERVICE_NAME, API_VERSION, credentials = credentials)
+
+def initialize_upload(youtube, file, title='Test Title', description=None,
+                      category=None, tags=None, privacy_status='private',
+                      recording_date=None, license=None):
+    """
+    Parameters
+    ----------
+    tags : list of str
+    """
+    metadata = dict(title=title)
+    if description:
+        metadata['description'] = description
+    if tags:
+        metadata['tags'] = tags
+    if category:
+        metadata['categoryId'] = category
+
+    body=dict(
+        snippet=metadata,
+        status=dict(privacyStatus=privacy_status)
+    )
+
+    if recording_date:
+        body['recordingDetails'] = dict(recordingDate=recording_date)
+    if license:
+        body['status']['license'] = license
+
+  # Call the API's videos.insert method to create and upload the video.
+    insert_request = youtube.videos().insert(
+        part=','.join(body.keys()),
+        body=body,
+        # The chunksize parameter specifies the size of each chunk of data, in
+        # bytes, that will be uploaded at a time. Set a higher value for
+        # reliable connections as fewer chunks lead to faster uploads. Set a lower
+        # value for better recovery on less reliable connections.
+        #
+        # Setting 'chunksize' equal to -1 in the code below means that the entire
+        # file will be uploaded in a single HTTP request. (If the upload fails,
+        # it will still be retried where it left off.) This is usually a best
+        # practice, but if you're using Python older than 2.6 or if you're
+        # running on App Engine, you should set the chunksize to something like
+        # 1024 * 1024 (1 megabyte).
+        media_body=MediaFileUpload(file, chunksize=-1, resumable=True)
+    )
+
+    return resumable_upload(insert_request)
+
+# This method implements an exponential backoff strategy to resume a
+# failed upload.
+def resumable_upload(request):
+    response = None
+    error = None
+    retry = 0
+    while response is None:
+        try:
+            print('Uploading file...')
```

Comment:
We probably shouldn’t have straight-up `print` statements in here. (I think they came from the YouTube example?) We should either…

1. Just drop them entirely.
2. Replace them with `debug()` calls like in `playlists.py`.
3. Use `logger.info` so you can turn the output on and off from outside this module. (Python docs here: https://docs.python.org/3.7/library/logging.html and you can check out web-monitoring-processing for example usage.)